MAUI/DataGrid/Clipboard-Operations.md

```diff
@@ -0,0 +1,546 @@
+        
+        clipBoardText = Regex.Split(clipBoardText[0], @"\t");
+
+        //Get the clipBoardText and check if the clipBoardText is more than one cell means call the base.
+
+        if (clipBoardText.Count() > 1)
+        {
+            base.OnPasteToCell(record, column, value);
+        }
+        
+        //Get the selectedCells for paste the copied cell 
+        var selectedCells = this.dataGrid.GetSelectedCells();
+        int selectedCellsCount = selectedCells.Count;
+        
+        for (int i = 0; i < selectedCellsCount; i++)
+        {
+            record = selectedCells[i].RowData;
+            
+            column = selectedCells[i].Column;
+            
+            //Call OnPasteToCell method with particular record of selectedCells,
+
+            //Column of selected records and rowData
+
+            if (record != null && column != null)
+                base.OnPasteToCell(record, column, value);
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+### Paste a record into many rows in Maui DataGrid
+
+By default, you can able to copy one row and paste it into another row when Row Selection is enabled in SfDataGrid. The below code shows how to copy one row and paste it into all selected rows by overriding the OnPasteToRow method in the DataGridCopyPasteController class.
+
+{% tabs %}
+{% highlight c# %}
+public class CustomCopyPaste : DataGridCopyPasteController
+{
+
+    public CustomCopyPaste(SfDataGrid dataGrid) : base(dataGrid)
+    {
+    }
+
+    protected override void OnPasteToRow(object clipBoardContent, object selectedRecords)
+    {
+        var text = Clipboard.GetTextAsync();
+        string[] clipBoardText = Regex.Split(text.ToString(), @"\r\n");
+
+        //Get the clipBoardText and check if the clipBoardText is more than one row means call the base.
+
+        if (clipBoardText.Count() > 1)
+        {
+            base.OnPasteToRow(clipBoardContent, selectedRecords);
+            return;
+        }
+
+        var selectedRecord = this.dataGrid.SelectedRows;
+        
+        for (int i = 0; i < selectedRecord.Count; i++)
+        {
+            //Get the selected records for paste the copied row.
+            selectedRecords = selectedRecord[i];
+            
+            //Call the OnPasteToRow method with clipBoardContent and selectedRecords
+            base.OnPasteToRow(clipBoardContent, selectedRecords);
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+### Select pasted records in Maui DataGrid
+
+By default after pasting the clipboard value to SfDataGrid selection is maintains in previously selected records as it is. The below code shows select the pasted records after the Paste operation, by overriding the OnPasteToRows and OnPasteToRow methods in DataGridCopyPasteController class. This code is applicable when SelectionUnit is [Row](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionUnit.html#Syncfusion_Maui_DataGrid_DataGridSelectionUnit_Row).
```

Comment:
By default, after pasting clipboard values to SfDataGrid, the selection remains on the previously selected records. The code below shows how to select the pasted records after the paste operation by overriding the `OnPasteToRows` and `OnPasteToRow` methods in the `DataGridCopyPasteController` class. This code is applicable when SelectionUnit is [Row](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionUnit.html#Syncfusion_Maui_DataGrid_DataGridSelectionUnit_Row).